lib/ansible/modules/cloud/ovh/ovh_dns.py

```diff
@@ -0,0 +1,344 @@
+        required: true
+        description:
+            - Name of the domain zone
+    subdomain:
+        default: ""
+        required: false
+        description:
+            - >
+              Subdomain of the DNS record. It has to be relative to your domain.
+              Leaving subdomain empty or unspecified will affect the domain
+              itself.
+    target:
+        required: true
+        description:
+            - Target of the DNS record.
+    type:
+        default: A
+        choices: ['A', 'AAAA', 'CNAME', 'DKIM', 'LOC', 'MX', 'NAPTR', 'PTR', 'SPF', 'SRV', 'SSHFP']
+        description:
+            - Type of DNS record (A, AAAA, PTR, CNAME, etc.)
+    ttl:
+        default: 0
+        description:
+            - Time to live associated to the DNS record. It's not mandatory
+              when deleting records therefore you could use default value 0,
+              but otherwise you shall pass a value. Also note that OVH does
+              not accept too low TTL values (< 60 seconds).
+    state:
+        default: present
+        choices: ['present', 'absent']
+        description:
+            - Determines wether the record is to be created/modified or deleted
+    endpoint:
+        required: true
+        description:
+            - The endpoint to use ( for instance ovh-eu)
+    application_key:
+        required: true
+        description:
+            - The applicationKey to use
+    application_secret:
+        required: true
+        description:
+            - The application secret to use
+    consumer_key:
+        required: true
+        description:
+            - The consumer key to use
+'''
+
+EXAMPLES = '''
+# Create an A record for "mydomain.com", pointing to 1.2.3.4
+- ovh_dns:
+    state: present
+    domain: mydomain.com
+    type: A
+    target: 1.2.3.4
+    ttl: 3600
+    endpoint: ovh-eu
+    application_key: yourkey
+    application_secret: yoursecret
+    consumer_key: yourconsumerkey
+
+# Create a A record for subdomain "api.staging.mydomain.com", pointing to 1.2.3.4
+- ovh_dns:
+    state: present
+    domain: mydomain.com
+    subdomain: db1.staging
+    target: 1.2.3.4
+    ttl: 3600
+    endpoint: ovh-eu
+    application_key: yourkey
+    application_secret: yoursecret
+    consumer_key: yourconsumerkey
+
+# Create a CNAME record "www" pointing to "cdn"
+- ovh_dns:
+    state: present
+    domain: mydomain.com
+    subdomain: www
+    type: CNAME
+    target: cdn
+    ttl: 3600
+    endpoint: ovh-eu
+    application_key: yourkey
+    application_secret: yoursecret
+    consumer_key: yourconsumerkey
+
+# Delete an existing record, must specify all parameters (TTL can be omitted)
+- ovh_dns:
+    state: absent
+    domain: mydomain.com
+    subdomain: www
+    type: CNAME
+    target: cdn
+    endpoint: ovh-eu
+    application_key: yourkey
+    application_secret: yoursecret
+    consumer_key: yourconsumerkey
+'''
+
+RETURN = '''
+'''
+
+ANSIBLE_METADATA = {'status': ['preview'],
```

Comment:
Remove this as it is already declared above.

Reply:
Now it makes sense... :man_facepalming: Thank you!